fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/runtime/DefaultApplicationPreparer.java

```diff
@@ -0,0 +1,119 @@
+package org.wildfly.swarm.jaxrs.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ArchiveEvent;
+import org.jboss.shrinkwrap.api.ArchiveEventHandler;
+import org.jboss.shrinkwrap.api.ArchivePath;
+import org.jboss.shrinkwrap.api.Node;
+import org.jboss.shrinkwrap.api.asset.ArchiveAsset;
+import org.jboss.shrinkwrap.api.asset.Asset;
+import org.jboss.shrinkwrap.api.asset.ByteArrayAsset;
+import org.objectweb.asm.ClassReader;
+import org.wildfly.swarm.config.runtime.AttributeDocumentation;
+import org.wildfly.swarm.jaxrs.JAXRSArchive;
+import org.wildfly.swarm.jaxrs.internal.ApplicationFactory2;
+import org.wildfly.swarm.jaxrs.internal.ApplicationPathAnnotationSeekingClassVisitor;
+import org.wildfly.swarm.spi.api.ArchivePreparer;
+import org.wildfly.swarm.spi.api.Defaultable;
+import org.wildfly.swarm.spi.api.annotations.Configurable;
+
+import static org.wildfly.swarm.spi.api.Defaultable.string;
+
+@ApplicationScoped
+public class DefaultApplicationPreparer implements ArchivePreparer {
+
+    @Configurable("swarm.jaxrs.application-path")
+    @AttributeDocumentation("Set the JAX-RS application path")
```

Comment:
I tend to be picky about these things, so sorry about that... This only has effect if the JAX-RS application class is autogenerated, right? If there's one provided by the user, this setting has no effect. The comment should reflect that. Something like "Set the JAX-RS application path, if the JAX-RS application class is autogenerated by Swarm".